### PR TITLE
implement step granularity

### PR DESCRIPTION
**Pull request by Trass3r**
https://microsoft.github.io/debug-adapter-protocol/specification#Types_SteppingGranularity

Also see https://github.com/microsoft/vscode/issues/102236.

### CODE REVIEW

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 1]
```diff
@@ -1302,12 +1304,24 @@ protected override void HandleContinueRequestAsync(IRequestResponder<ContinueArg
...
         {
-            StepInResponse response = new StepInResponse();
+            try
+            {
+                var granularity = responder.Arguments.Granularity.GetValueOrDefault();
+                StepInternal(responder.Arguments.ThreadId, enum_STEPKIND.STEP_INTO, granularity, AD7Resources.Error_Scenario_Step_In);
+                responder.SetResponse(new StepInResponse());
+            }
+            catch (AD7Exception e)
+            {
```
Comment (gregg-miskelly) [inline src/OpenDebugAD7/AD7DebugSession.cs:1314]: >{ [](http://example.com/codeflow?start=12&length=1)

Shouldn't this have `responder.SetError(new ProtocolException(e.Message));`?

**Review by gregg-miskelly — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: src/OpenDebugAD7/AD7DebugSession.cs (hunk 1 above)
Comment (Trass3r) [reply src/OpenDebugAD7/AD7DebugSession.cs:1314]: Thx done!

**Review by gregg-miskelly — APPROVED** [review]

**Review by WardenGnaw — APPROVED** [review]

**Comment by WardenGnaw** [comment]
Not sure why the checks are not running. Investigating.

**Comment by Trass3r** [comment]
Indeed they haven't been run for a while.

**Comment by WardenGnaw** [comment]
It seems like AzureDevOps is not correctly working with branches not in the Microsoft/MIEngine repo. I'll see if we can migrate to GitHub Actions which seems to be working correctly.

**Pull request merged**